### PR TITLE
[UIKit] Add support for Xcode 15.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/uikit.cs [hunk 1]
```diff
@@ -26761,5 +27283,1203 @@ interface UITextInputContext {
...
+	[NoWatch, NoTV, iOS (17, 0), MacCatalyst (17, 0)]
+#if XAMCORE_5_0
+	[BaseType (typeof(UIViewController))]
+	interface UIDocumentViewController
+#else
+	[BaseType (typeof (UIViewController), Name = "UIDocumentViewController")]
+	interface UIDocViewController
+#endif
+	{
+
```
Comment (haritha-mohan) [inline src/uikit.cs:27558]:
```suggestion
```

File: src/uikit.cs [hunk 2]
```diff
@@ -26761,5 +27283,1203 @@ interface UITextInputContext {
...
+		[Export ("shapeByApplyingInset:")]
+		UIResolvedShape CreateShape (nfloat inset);
+	}
+
+	interface IUIHoverEffect { }
+
+	[NoWatch, NoTV, iOS (17, 0), MacCatalyst (17, 0)]
+	[Protocol]
+	interface UIHoverEffect { }
```
Comment (haritha-mohan) [inline src/uikit.cs:27673]:
```suggestion
	[NoWatch, NoTV, iOS (17, 0), MacCatalyst (17, 0)]
#if NET
	[Protocol, Model]
#else
	[Protocol, Model (AutoGeneratedName = true)]
#endif
	[BaseType (typeof (NSObject))]
	interface UIHoverEffect { }
```

File: src/uikit.cs [hunk 3]
```diff
@@ -26761,5 +27283,1203 @@ interface UITextInputContext {
...
+		[Export ("defaultReader", ArgumentSemantic.Strong)]
+		UIImageReader DefaultReader { get; }
+
+		[Static]
+		[Export ("readerWithConfiguration:")]
+		UIImageReader CreateReader (UIImageReaderConfiguration configuration);
+
+		[Export ("configuration", ArgumentSemantic.Copy)]
+		UIImageReaderConfiguration Configuration { get; }
+		// -(UIImage * _Nonnull)imageWithContentsOfFileURL:(NSURL * _Nonnull)url;
```
Comment (haritha-mohan) [inline src/uikit.cs:27770]: not sure if this was left for additional context or by accident
```suggestion

```

File: src/uikit.cs [hunk 4]
```diff
@@ -26761,5 +27283,1203 @@ interface UITextInputContext {
...
+
+		[Export ("undoRedoItemGroup")]
+		UIBarButtonItemGroup UndoRedoItemGroup { get; }
+	}
+
+	interface IUIShapeProvider { }
+
+	[NoWatch, NoTV, iOS (17, 0), MacCatalyst (17, 0)]
+	[Protocol]
+	interface UIShapeProvider {
```
Comment (haritha-mohan) [inline src/uikit.cs:27588]: iirc protocols need both model and base type attributes?
```suggestion
	[NoWatch, NoTV, iOS (17, 0), MacCatalyst (17, 0)]
#if NET
	[Protocol, Model]
#else
	[Protocol, Model (AutoGeneratedName = true)]
#endif
	[BaseType (typeof (NSObject))]
	interface UIShapeProvider {
```

File: src/uikit.cs [hunk 5]
```diff
@@ -26761,5 +27283,1203 @@ interface UITextInputContext {
...
+		void MoveToPoint (CGPoint point, CGRect caretRect, bool tracksCaret);
+
+		[Export ("invalidate")]
+		void Invalidate ();
+	}
+
+	interface IUITextCursorView { }
+
+	[NoWatch, TV (17, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[Protocol]
```
Comment (haritha-mohan) [inline src/uikit.cs:28007]: missing base type/model?

File: src/uikit.cs [hunk 6]
```diff
@@ -26761,5 +27283,1203 @@ interface UITextInputContext {
...
+
+		[Abstract]
+		[Export ("resetBlinkAnimation")]
+		void ResetBlinkAnimation ();
+	}
+
+	interface IUITextSelectionHighlightView { }
+
+	[NoWatch, TV (17, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[Protocol]
```
Comment (haritha-mohan) [inline src/uikit.cs:28021]: missing base type/model?

File: src/uikit.cs [hunk 7]
```diff
@@ -26761,5 +27283,1203 @@ interface UITextInputContext {
...
+	interface UITextSelectionHighlightView : UICoordinateSpace {
+		[Abstract]
+		[Export ("selectionRects", ArgumentSemantic.Strong)]
+		UITextSelectionRect [] SelectionRects { get; set; }
+	}
+
+	interface IUITextSelectionHandleView { }
+
+	[NoWatch, TV (17, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[Protocol]
```
Comment (haritha-mohan) [inline src/uikit.cs:28031]: there's a few more below that need to be addressed

**Review by haritha-mohan — CHANGES_REQUESTED** [review]

File: src/uikit.cs [hunk 8]
```diff
@@ -8830,6 +8954,25 @@ interface UIImageConfiguration : NSCopying, NSSecureCoding {
...
+		[Static]
+		[Watch (10, 0), TV (17, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("configurationWithLocale:")]
+		UIImageConfiguration FromConfiguration ([NullAllowed] NSLocale locale);
+
+		[NoWatch, TV (17, 0), iOS (17, 0)]
+		[Static]
+		[Export ("configurationWithTraitCollection:")]
+		UIImageConfiguration ConfigurationWithTraitCollection ([NullAllowed] UITraitCollection traitCollection);
+
```
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply src/uikit.cs:8975]: Doesn't really matter but adding.

File: src/uikit.cs (hunk 5 above)
Comment (mandel-macaque) [reply src/uikit.cs:28007]: Same, this is not going to be implemented by users as opposed to the [UITextSelectionDisplayInteractionDelegate](https://developer.apple.com/documentation/uikit/uitextselectiondisplayinteractiondelegate?language=objc) which does contain Model in it.

**Review by haritha-mohan — APPROVED** [review]

File: tests/xtro-sharpie/api-annotations-dotnet/iOS-UIKit.ignore [hunk 9]
```diff
@@ -17,12 +17,83 @@
 ## the category name is clear enough: UIPrintPaper_Deprecated_Nonfunctional
 !missing-selector! UIPrintPaper::printRect not bound
 
-## unsorted
+# Added in UIAccessibility, this is later inlined in UIView, else we will
+# add all of the properties to all objects whihc is not correct.
```
Comment (tj-devel709) [inline tests/xtro-sharpie/api-annotations-dotnet/iOS-UIKit.ignore:21]:
```suggestion
# add all of the properties to all objects which is not correct.
```

**Review by tj-devel709 — APPROVED** [review]
Only spotted a really tiny typo but other than that LGTM

File: src/uikit.cs [hunk 10]
```diff
@@ -8830,6 +8954,24 @@ interface UIImageConfiguration : NSCopying, NSSecureCoding {
...
+		NSLocale Locale { get; }
+
+		[Watch (10, 0), TV (17, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("configurationWithLocale:")]
+		UIImageConfiguration GetConfiguration ([NullAllowed] NSLocale locale);
+
+		[Static]
+		[Watch (10, 0), TV (17, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("configurationWithLocale:")]
+		UIImageConfiguration FromConfiguration ([NullAllowed] NSLocale locale);
```
Comment (rolfbjarne) [inline src/uikit.cs:8969]:
```suggestion
		UIImageConfiguration CreateConfiguration ([NullAllowed] NSLocale locale);
```

File: src/uikit.cs [hunk 11]
```diff
@@ -8830,6 +8954,24 @@ interface UIImageConfiguration : NSCopying, NSSecureCoding {
...
+
+		[NoWatch, TV (17, 0), iOS (17, 0)]
+		[Static]
+		[Export ("configurationWithTraitCollection:")]
+		UIImageConfiguration ConfigurationWithTraitCollection ([NullAllowed] UITraitCollection traitCollection);
```
Comment (rolfbjarne) [inline src/uikit.cs:8974]:
```suggestion
		UIImageConfiguration CreateConfiguration ([NullAllowed] UITraitCollection traitCollection);
```

File: src/uikit.cs [hunk 12]
```diff
@@ -16758,6 +17032,24 @@ interface UIView : UIAppearance, UIAppearanceContainer, UIAccessibility, UIDynam
...
+		IUITraitOverrides TraitOverrides { get; }
+
+		[Watch (10, 0), TV (17, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("updateTraitsIfNeeded")]
+		void UpdateTraitsIfNeeded ();
+
+		[NoWatch, NoTV, iOS (17, 0), MacCatalyst (17, 0)]
+		[NullAllowed, Export ("hoverStyle", ArgumentSemantic.Copy)]
+		UIHoverStyle HoverStyle { get; set; }
+		[Async]
```
Comment (rolfbjarne) [inline src/uikit.cs:17048]: Missing line:

```suggestion

		[Async]
```

File: src/uikit.cs [hunk 13]
```diff
@@ -17822,6 +18165,86 @@ partial interface UITraitCollection : NSCopying, NSSecureCoding {
...
+		nint GetValueForTrait (IUINSIntegerTraitDefinition trait);
+
+		[Watch (10, 0), TV (17, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Static]
+		[Export ("traitCollectionWithObject:forTrait:")]
+		UITraitCollection GetTraitCollectionWithObject ([NullAllowed] NSObject @object, IUIObjectTraitDefinition trait);
+
+		[Watch (10, 0), TV (17, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("traitCollectionByReplacingObject:forTrait:")]
+		UITraitCollection TraitCollectionByReplacingObject ([NullAllowed] NSObject @object, IUIObjectTraitDefinition trait);
```
Comment (rolfbjarne) [inline src/uikit.cs:18210]:
```suggestion
		UITraitCollection GetTraitCollectionByReplacingObject ([NullAllowed] NSObject @object, IUIObjectTraitDefinition trait);
```

File: src/uikit.cs [hunk 14]
```diff
@@ -26761,5 +27280,1221 @@ interface UITextInputContext {
...
+		[Export ("defaultReader", ArgumentSemantic.Strong)]
+		UIImageReader DefaultReader { get; }
+
+		[Static]
+		[Export ("readerWithConfiguration:")]
+		UIImageReader CreateReader (UIImageReaderConfiguration configuration);
+
+		[Export ("configuration", ArgumentSemantic.Copy)]
+		UIImageReaderConfiguration Configuration { get; }
+		[Export ("imageWithContentsOfFileURL:")]
```
Comment (rolfbjarne) [inline src/uikit.cs:27766]: Missing line:

```suggestion

		[Export ("imageWithContentsOfFileURL:")]
```

File: src/uikit.cs [hunk 15]
```diff
@@ -26761,5 +27280,1221 @@ interface UITextInputContext {
...
+	[TV (17, 0), iOS (17, 0), MacCatalyst (17, 0), NoWatch]
+#if NET
+	[Protocol, Model]
+#else
+	[Protocol, Model (AutoGeneratedName = true)]
+#endif
+	[BaseType (typeof (NSObject))]
+	interface UIPageControlProgressDelegate {
+		[Export ("pageControlProgress:initialProgressForPage:")]
+		float InitialProgressForPage (UIPageControlProgress progress, nint page);
```
Comment (rolfbjarne) [inline src/uikit.cs:27808]:
```suggestion
		float GetInitialProgressForPage (UIPageControlProgress progress, nint page);
```

File: src/uikit.cs [hunk 16]
```diff
@@ -26761,5 +27280,1221 @@ interface UITextInputContext {
...
+#if NET
+	[Protocol, Model]
+#else
+	[Protocol, Model (AutoGeneratedName = true)]
+#endif
+	[BaseType (typeof (NSObject))]
+	interface UITextSelectionDisplayInteractionDelegate {
+		[Export ("selectionContainerViewBelowTextForSelectionDisplayInteraction:")]
+		[return: NullAllowed]
+		UIView GetSelectionContainerViewBellowText (UITextSelectionDisplayInteraction interaction);
```
Comment (rolfbjarne) [inline src/uikit.cs:28094]:
```suggestion
		UIView GetSelectionContainerViewBelowText (UITextSelectionDisplayInteraction interaction);
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Review by dalexsoto — APPROVED** [review]
Just fix the breaking change please!

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**